src/BuildingRegistry.Api.BackOffice/BuildingUnit/BuildingUnitController-Plan.cs

```diff
@@ -0,0 +1,80 @@
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Abstractions.Building.Requests;
+    using Abstractions.Building.Validators;
+    using Abstractions.BuildingUnit.Requests;
+    using Be.Vlaanderen.Basisregisters.AggregateSource;
+    using Be.Vlaanderen.Basisregisters.Api.Exceptions;
+    using Building;
+    using BuildingRegistry.Building.Exceptions;
+    using FluentValidation;
+    using FluentValidation.Results;
+    using Handlers;
+    using Infrastructure.Options;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Options;
+    using Swashbuckle.AspNetCore.Filters;
+
+    public partial class BuildingUnitController
+    {
+        /// <summary>
+        /// Plan een gebouweenheid in.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="request"></param>
+        /// <param name="validator"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [HttpPost("acties/plannen")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        [SwaggerResponseHeader(StatusCodes.Status202Accepted, "location", "string", "De url van de geplande gebouweenheid.")]
+        [SwaggerRequestExample(typeof(PlanBuildingRequest), typeof(PlanBuildingRequestExamples))]
+        [SwaggerResponseExample(StatusCodes.Status400BadRequest, typeof(BadRequestResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status500InternalServerError, typeof(InternalServerErrorResponseExamples))]
+
+        public async Task<IActionResult> Plan(
+            [FromServices] IOptions<ResponseOptions> options,
+            [FromServices] IValidator<PlanBuildingUnitRequest> validator,
+            [FromBody] PlanBuildingUnitRequest request,
+            CancellationToken cancellationToken = default)
+        {
+            await validator.ValidateAndThrowAsync(request, cancellationToken);
+
+            try
+            {
+                request.Metadata = GetMetadata();
+                var response = await _mediator.Send(request, cancellationToken);
+
+                return new AcceptedWithETagResult(
+                    new Uri(string.Format(options.Value.BuildingUnitDetailUrl, response.BuildingUnitPersistentLocalId)),
+                    response.LastEventHash);
+            }
+            catch (IdempotencyException)
+            {
+                return Accepted();
+            }
+            catch (DomainException exception)
+            {
+                throw exception switch
+                {
+                    BuildingUnitNotFoundException => new ApiException(ValidationErrorMessages.BuildingNotFound, StatusCodes.Status404NotFound),
```

Comment:
`BuildingNotFound` ? Does a test not cover this?

Reply:
If a building is not found, it cannot be a 404, but a 400. So a validation exception.